segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -870,6 +864,38 @@ private void persistMetadata(LogMetadata metadata, boolean create) throws Durabl
         log.info("{} Metadata persisted ({}).", this.traceObjectId, metadata);
     }
 
+    @VisibleForTesting
+    protected Stat createZkMetadata(byte[] serializedMetadata) throws Exception {
+        val result = new Stat();
+        this.zkClient.create().creatingParentsIfNeeded().storingStatIn(result).forPath(this.logNodePath, serializedMetadata);
+        return result;
+    }
+
+    @VisibleForTesting
+    protected Stat updateZkMetadata(byte[] serializedMetadata, int version) throws Exception {
+        return this.zkClient.setData().withVersion(version).forPath(this.logNodePath, serializedMetadata);
+    }
+
+    /**
+     * Verifies the given {@link LogMetadata} against the actual one stored in ZooKeeper.
+     *
+     * @param metadata The Metadata to check.
+     * @return True if the metadata stored in ZooKeeper is an identical match to the given one, false otherwise. If true,
+     * {@link LogMetadata#getUpdateVersion()} will also be updated with the one stored in ZooKeeper.
+     */
+    private boolean reconcileMetadata(LogMetadata metadata) {
+        try {
+            val actualMetadata = loadMetadata();
```

Comment:
there is a case in which `loadMetadata()` can return `null`. Maybe this is handled by the `equals` generated by Lombok, but just wanted to create awareness on this corner case.